internal/component/pyroscope/write/write.go

```diff
@@ -466,3 +477,39 @@ func ensureNameMatchesService(lbls labels.Labels) labels.Labels {
 	}
 	return lbls
 }
+
+// validateLabels checks for valid labels and doesn't contain duplicates.
+func validateLabels(lbls labels.Labels) error {
+	if lbls.Len() == 0 {
+		return labelset.ErrServiceNameIsRequired
+	}
+
+	sort.Sort(lbls)
+
+	lastLabelName := ""
+	for _, l := range lbls {
+		if cmp := strings.Compare(lastLabelName, l.Name); cmp == 0 {
+			return fmt.Errorf("duplicate label name: %s", l.Name)
+		}
+
+		// Validate label value
+		if !model.LabelValue(l.Value).IsValid() {
+			return fmt.Errorf("invalid label value for %s: %s", l.Name, l.Value)
+		}
+
+		// Skip label name validation for pyroscope reserved labels
+		if l.Name == pyroscope.LabelName {
+			lastLabelName = l.Name
+			continue
+		}
+
+		// Validate label name
+		if err := labelset.ValidateLabelName(l.Name); err != nil {
+			return fmt.Errorf("invalid label name: %w", err)
+		}
```

Comment:
Nit: Find that easier to read and less error prone

```suggestion
		// Skip label name validation for pyroscope reserved labels
		if l.Name != pyroscope.LabelName {
  		  		// Validate label name
		  		if err := labelset.ValidateLabelName(l.Name); err != nil {
		  		  		return fmt.Errorf("invalid label name: %w", err)
		  		  }
		}

```

Reply:
If you disagree, just ignore